py/obj.c

```diff
@@ -355,9 +355,13 @@ bool mp_obj_get_float_maybe(mp_obj_t arg, mp_float_t *value) {
     } else if (mp_obj_is_float(arg)) {
         val = mp_obj_float_get(arg);
     } else {
-        return false;
+        arg = mp_unary_op(MP_UNARY_OP_FLOAT, (mp_obj_t)arg);
```

Comment:
ok, couple of problems here - `mp_unary_op` raises exception on fail so doesn't work as-is for a `_maybe` conversion function. 

I could move it to `mp_obj_get_float()` but that would then take over the error message in the exception which isn't correct in the general case.

Is there something I'm forgetting here?

Perhaps it's worth splitting `mp_unary_op()` out to also provide a `mp_unary_op_maybe()` function?

Reply:
We could make it `MP_UNARY_OP_FLOAT_MAYBE`, and then add a check towards the end of `mp_unary_op()`, after the `op == MP_UNARY_OP_BOOL` check:
```c
if (op == MP_UNARY_OP_FLOAT_MAYBE) {
    return MP_OBJ_NULL;
}
```

Reply:
For some reason that feels a little wrong to me... not sure why. It's probably going to be quite clean and efficient, better than splitting unary into two functions that doubles up the stack calls for all other users. I'll try it out!

Reply:
In practice I quite like this approach, it works quite nicely.